lib/crossing.rb

```diff
@@ -9,26 +9,27 @@ def initialize(s3_client)
   end
 
   def put(bucket, filename)
-    begin
-      file = File.new(filename, 'r')
-    rescue
-      raise CrossingFileNotFoundException, "File not found: #{filename}"
+    File.open(filename, 'r') do |file|
```

Comment:
If you pass a block to File.open, then the #close of the filehandle is done automatically when the block ends